### PR TITLE
Add menu in the tree editor

### PR DESCRIPTION
**Pull request by martinRenou**

### CODE REVIEW

**Comment by martinRenou** [comment]
Needs tests

File: force_wfmanager/left_side_pane/workflow_settings.py [hunk 1]
```diff
@@ -32,6 +36,66 @@ def get_text(self, object, trait, index):
...
+        """ Opens a dialog for creating the KPI Calculator """
+
+    def delete_mco_handler(self, editor, object):
+        object.model.multi_criteria_optimizer = None
+
+    def delete_mco_is_enabled(self, editor, object):
+        return object.model.multi_criteria_optimizer is not None
+
+    def delete_data_sources_handler(self, editor, object):
+        object.model.data_sources = []
```
Comment (stefanoborini) [inline force_wfmanager/left_side_pane/workflow_settings.py]: [:]

File: force_wfmanager/left_side_pane/workflow_settings.py [hunk 2]
```diff
@@ -32,6 +36,78 @@ def get_text(self, object, trait, index):
         return get_bundle_name(self.item)
 
 
+class TreeEditorHandler(Handler):
+    def new_mco_handler(self, editor, object):
```
Comment (martinRenou) [inline force_wfmanager/left_side_pane/workflow_settings.py]: do not use object. it is a "protected" name.

File: force_wfmanager/left_side_pane/workflow_settings.py [hunk 3]
```diff
@@ -32,6 +36,78 @@ def get_text(self, object, trait, index):
...
+new_kpi_calculator_action = Action(
+    name='New KPI Calculator...',
+    action='handler.new_kpi_calculator_handler(editor, object)')
+
+delete_mco_action = Action(
+    name='Delete',
+    action='handler.delete_mco_handler(editor, object)'
+)
+
+delete_data_sources_action = Action(
```
Comment (martinRenou) [inline force_wfmanager/left_side_pane/workflow_settings.py]: remove it.

File: test/test_workflow_settings.py [hunk 4]
```diff
@@ -154,3 +195,110 @@ def test_get_bundle_name(self):
...
         self.assertEqual(
             get_bundle_name(unnamed_bundle), 'enthought.test.bundle.unnamed')
+
+
+class TestTreeEditorHandler(unittest.TestCase):
+    def setUp(self):
+        self.handler = TreeEditorHandler()
+
+    def test_delete_data_sources_is_disabled(self):
+        emptyWorkflow = get_empty_workflow_model_view()
```
Comment (martinRenou) [inline test/test_workflow_settings.py]: name

File: test/test_workflow_settings.py [hunk 5]
```diff
@@ -154,3 +195,110 @@ def test_get_bundle_name(self):
...
+        self.assertFalse(
+            self.handler.delete_data_sources_is_enabled(
+                None, emptyWorkflow)
+        )
+
+    def test_delete_data_sources_is_enabled(self):
+        filledWorkflow = get_filled_workflow_model_view()
```
Comment (martinRenou) [inline test/test_workflow_settings.py]: name

File: test/test_workflow_settings.py [hunk 6]
```diff
@@ -154,3 +195,110 @@ def test_get_bundle_name(self):
         unnamed_bundle = UnnamedBundle()
         self.assertEqual(
             get_bundle_name(unnamed_bundle), 'enthought.test.bundle.unnamed')
+
+
+class TestTreeEditorHandler(unittest.TestCase):
+    def setUp(self):
+        self.handler = TreeEditorHandler()
+
```
Comment (martinRenou) [inline test/test_workflow_settings.py]: self.filled_workflow=...

File: force_wfmanager/left_side_pane/workflow_settings.py (hunk 2 above)
Comment (martinRenou) [inline force_wfmanager/left_side_pane/workflow_settings.py]: `object` is passed as a keyword argument inside traitsui magic, so unfortunately we do not have control on this name

**Pull request merged**